### PR TITLE
[hma][scripts] Add API methods to script utils

### PR DESCRIPTION
**Pull request by BarrettOlson**
Summary
---------

Expands `HasherMatcherActionerAPI` to support a number of API endpoints useful for testing. 

Sneaky side fixes (can breakout to own PR if requested):
- Remove vestigial `MatchSummary` field
- Fixed an issue in `authentication-shared/main.tf` and adds to `authentication/main.tf` so `get_auth_token` will work for all deployments with the right local aws credentials.

Test Plan
---------

Manually confirmed all new methods work (see bottom of `scripts/script_utils.py`) EXCEPT for `create_action` and `create_action_rule` which I plan to test as part of 'test deployments script' in my next PR.

### CODE REVIEW

File: hasher-matcher-actioner/scripts/script_utils.py [hunk 1]
```diff
@@ -30,14 +62,38 @@ def _get_header(self) -> t.Dict[str, str]:
     def _get_request_url(self, api_path: str) -> str:
         return urljoin(self.api_url, api_path)
 
+    def _get_with_auth(
```
Comment (schatten) [inline hasher-matcher-actioner/scripts/script_utils.py:65]: A more flexible way to do this is to expose a `requests.Session` object [1]. That way users have access to the full gamut of requests capabilities while re-using the auth. A session object has almost all `request.*` methods available.

Imagine having to write `_post_multi_part_with_auth()` and `_put_with_auth()` and `_delete_with_auth()`

[1]: https://docs.python-requests.org/en/master/user/advanced/

Comment (schatten) [reply hasher-matcher-actioner/scripts/script_utils.py:65]: Oh, and did I mention it respects Keep-Alive and re-uses the underlying TCP connection for higher performance?

File: hasher-matcher-actioner/scripts/script_utils.py [hunk 2]
```diff
@@ -86,7 +134,137 @@ def send_single_submission_url(
             headers={"content-type": "image/jpeg"},
         )
 
+    def get_content_hash_details(
+        self,
+        content_id: str,
+        api_path: str = "/content/hash/",
+    ):
```
Comment (schatten) [inline hasher-matcher-actioner/scripts/script_utils.py:141]: typing? Can we re-use the representation classes from hmalib here? I guess that is what you were planning to do. Totally okay to do that as separate PR.

Comment (schatten) [reply hasher-matcher-actioner/scripts/script_utils.py:141]: all new API methods in this module seem to lack that. You can address my concerns as a reply to this message.

**Review by schatten — CHANGES_REQUESTED** [review]
Would prefer using requests.Session instead of creating proxy methods. If you do it now, adding DELETE and PUT calls will be simpler and have the full array of requests' feature set.

Other than that, nit about typing. You mentioned that you'll use hmalib as a dependency. If we can return objects of the representation classes (return types in API methods), that would reduce the need to refer to attributes in the objects. eg. 
```python
        return response.json().get("action_history", [])
```

File: hasher-matcher-actioner/scripts/script_utils.py (hunk 2 above)
Comment (BarrettOlson) [reply hasher-matcher-actioner/scripts/script_utils.py:141]: I am intentionally avoiding having this class import hmalib so for things like `scripts/storm_submissions_api` we only need this file atm. 

For now scripts that are deeply tied to representation classes from hmalib can import it themselves if/when that becomes impossible I will move them into this class as well. For the time being this file is structured more like `webapp/src/Api.jsx`

File: hasher-matcher-actioner/scripts/script_utils.py (hunk 1 above)
Comment (BarrettOlson) [reply hasher-matcher-actioner/scripts/script_utils.py]: sure! I was avoiding cookies but since we control the API and don't set any the concern is likely unfounded.

File: hasher-matcher-actioner/scripts/script_utils.py [hunk 3]
```diff
@@ -62,31 +100,169 @@ def send_single_submission_url(
...
         put_response = requests.put(
             response_json["presigned_url"],
             data=file,
             headers={"content-type": "image/jpeg"},
         )
 
+    def get_content_hash_details(
+        self,
+        content_id: str,
+        api_path: str = "/content/hash/",
```
Comment (schatten) [inline hasher-matcher-actioner/scripts/script_utils.py:124]: Why is the `api_path` an argument? Not seeing the utility here? Do we expect this to change?

**Review by schatten — APPROVED** [review]
Approving to unblock. Do not see why the URL is not internal to the method but can be overridden at callsites. More a question than an RC.

Other than that, LGTM.

File: hasher-matcher-actioner/scripts/script_utils.py (hunk 3 above)
Comment (BarrettOlson) [reply hasher-matcher-actioner/scripts/script_utils.py:124]: I found overriding helpful for debugging and testing. If I create `/content/new-hash/` for example that I want to function the same but not replace the old impl just yet.
It's roughly the same number of lines but if these methods become too much like Swiss Army Knife Anti-patterns I'll refactor.
